web/source/kmwdevice.ts

```diff
@@ -91,6 +91,7 @@ namespace com.keyman {
           if(!results) {
             console.warn("KMW could not properly parse the user agent string."
               + "A suboptimal keyboard layout may result.");
+            this.OS='MacOSX';
```

Comment:
Can't this go in the outer block?

Reply:
Unfortunately, no.  That'd break our iOS device detection whenever a user prefers Safari to show desktop pages, which is the iOS default setting on tablets.

There actually _is_ an override that can be set on the WebView, but we don't currently have that in place.  I'd personally prefer to have both in place, just as a failsafe should one or the other break.